### PR TITLE
Fix imposibility to deploy Aleo programs that have imports

### PR DESCRIPTION
**Pull request by ilitteri**
## Motivation

There was not possible to deploy Aleo programs with imports from the Aleo CLI.

## The problem

Only the main program was in the process when deploying.

## Solution

Added all the main program imports to the process before deploying.

## Related PRs

This issue was mentioned in [this comment](https://github.com/AleoHQ/aleo/pull/326#issuecomment-1227898092) in the Aleo repo.

### CODE REVIEW

File: vm/package/deploy.rs [hunk 1]
```diff
@@ -137,35 +138,46 @@ impl<N: Network> Package<N> {
     pub fn deploy<A: crate::circuit::Aleo<Network = N, BaseField = N::Field>>(
```
Comment (raychu86) [inline vm/package/deploy.rs:137]: The logic for the subsequent functions are quite straightforward, try to keep them all contained the the `deploy` function.

File: vm/package/deploy.rs [hunk 2]
```diff
@@ -137,35 +138,46 @@ impl<N: Network> Package<N> {
...
         let program = self.program();
+        let mut process = Process::<N>::load()?;
+        self.add_imported_programs_to_process(&mut process, program.imports().keys())?;
+        if let Some(ref endpoint) = endpoint {
+            Self::deploy_program::<A>(&process, program, endpoint)?;
+        }
+        Ok(())
+    }
 
-        // Retrieve the program ID.
```
Comment (raychu86) [inline vm/package/deploy.rs:144]: Try to document each step like it's done previously. For example:

```
pub fn deploy() {
// Retrieve the main program.
...

// Retrieve the program ID.
...

// Add the imported programs.

...

```

Comment (facundoolano) [reply vm/package/deploy.rs:144]: I actually suggested him not to do that 😅 I think it makes sense when the comment helps clarifying the code below, but they are distracting when they just repeat what the code already states, e.g.

``` 
// Retrieve the program ID.
let program_id = program.id();
```

Comment (raychu86) [reply vm/package/deploy.rs:144]: In general, we like to err on the side of over documenting, so when api's change, we still know the exact intentions of the logic.

File: vm/package/deploy.rs [hunk 3]
```diff
@@ -137,35 +137,40 @@ impl<N: Network> Package<N> {
     pub fn deploy<A: crate::circuit::Aleo<Network = N, BaseField = N::Field>>(
         &self,
         endpoint: Option<String>,
-    ) -> Result<Deployment<N>> {
```
Comment (raychu86) [inline vm/package/deploy.rs:140]: If we remove this `Deployment` output, what is the purpose of the case where `endpoint` is None.

When would the user tries to call `deploy` without an endpoint? The deployment is never generated, but the function runs to completion without issue.

File: vm/package/deploy.rs (hunk 2 above)
Comment (facundoolano) [reply vm/package/deploy.rs:144]: Yeah I prefer to err on the side of over-documenting as well, but comments should complement or augment the information provided by the code, not reiterate it. E.g. establish intent, context and assumptions, especially at the interfaces (module, struct and function-level), and explaining/clarifying the complex parts of the implementation. From that perspective I'd say this module is under-documented even though it has many comments. Comments that just repeat the code are distracting to the reader and add maintenance burden without providing value.

**Pull request closed without merging**